packages/core/src/connectors/injected.ts

```diff
@@ -223,13 +223,13 @@ export function injected(parameters: InjectedParameters = {}) {
           currentChainId = chain?.id ?? currentChainId
         }
 
-        if (shimDisconnect) {
-          // Remove disconnected shim if it exists
+        // Remove disconnected shim if it exists
+        if (shimDisconnect)
           await config.storage?.removeItem(`${this.id}.disconnected`)
-          // Add connected shim if no target exists
-          if (!parameters.target)
-            await config.storage?.setItem('injected.connected', true)
-        }
+
+        // Add connected shim if no target exists
+        if (!parameters.target)
+          await config.storage?.setItem('injected.connected', true)
```

Comment:
flag needs to be set even if `shimDisconnect: false`

https://github.com/wevm/wagmi/blob/f612323f2dd9a80cc5be43f0bc9fd08c31aeb65c/packages/core/src/connectors/injected.ts#L316